socket.c

```diff
@@ -289,6 +289,7 @@ int cSatipSocket::ReadMulti(unsigned char *bufferAddrP, unsigned int *elementRec
      error("%s Invalid parameter(s)", __PRETTY_FUNCTION__);
      return -1;
      }
+#if defined(__GLIBC__)
 #if defined(__GLIBC_PREREQ) && __GLIBC_PREREQ(2,12)
```

Comment:
*sigh* IMO, it would make a better readability to add libc detection right after the boiler plate:
```
#if defined(__GLIBC__) // musl-libc evaluates the whole line at once
#if defined(__GLIBC_PREREQ) && __GLIBC_PREREQ(2,12)
#define ENABLE_RECVMMSG
#endif
#endif
...
#if defined(ENABLE_RECVMMSG)
#else
#endif
```

Another question: why don't you use the recvmmsg code path as musl-libc seems to support it?
https://git.musl-libc.org/cgit/musl/tree/src/network/recvmmsg.c